package/MDAnalysis/lib/_cutil.pyx

```diff
@@ -47,47 +59,951 @@ ctypedef cset[int] intset
+            center[i, 1] = NAN
+            center[i, 2] = NAN
+        else:
+            center[i, 0] *= inv_sum_weights
+            center[i, 1] *= inv_sum_weights
+            center[i, 2] *= inv_sum_weights
+    return zero_weights
+
+
+def unique_int_1d(np.intp_t[:] values not None, bint return_counts=False,
+                  bint return_masks=False, bint assume_unsorted=False):
     """Find the unique elements of a 1D array of integers.
 
     This function is optimal on sorted arrays.
 
     Parameters
     ----------
     values: numpy.ndarray
-        1D array of dtype ``numpy.int64`` in which to find the unique values.
+        1D array of dtype ``numpy.intp`` (or equivalent) in which to find the
+        unique values.
+    return_counts: bool, optional
+        If ``True``, the number of occurrences of each unique value is returned
+        as well.
+    return_masks: bool, optional
+        If ``True``, an array of masks (with one mask per unique value) will be
+        returned as well.
+    assume_unsorted: bool, optional
+        If `values` is known to be unsorted (i.e., its values are not
+        monotonically increasing), setting `assume_unsorted` to ``True`` can
+        speed up the computation.
 
     Returns
     -------
-    numpy.ndarray
+    unique: numpy.ndarray
         A deduplicated copy of `values`.
+    counts: numpy.ndarray, optional
+        An array of the same length as `unique` containing the number of
+        occurrences of each unique value in the original `values` array. Only
+        returned if `return_counts` is ``True``.
+    masks: numpy.ndarray, optional
+        An array of dtype ``object`` containing a mask for each value in
+        `unique`. Each of the masks allows accessing all occurrences of its
+        corresponding unique value in `values` such that
+        ``numpy.all(values[masks[i]] == unique[i]) == True``. Thus, the masks
+        array is roughly equivalent to
+        ``[numpy.where(values == i) for i in numpy.unique(values)]``. Only
+        returned if `return_masks` is ``True``.
+
+    Notes
+    -----
+    The dtype ``numpy.intp`` is usually equivalent to ``numpy.int32`` on a 32
+    bit operating system, and, likewise, equivalent to ``numpy.int64`` on a 64
+    bit operating system. The exact behavior is compiler-dependent and can be
+    checked with ``print(numpy.intp)``.
+
+
+    See Also
+    --------
+    :func:`numpy.unique`
+    :func:`unique_masks_int_1d`
 
 
     .. versionadded:: 0.19.0
+    .. versionchanged:: 0.20.0
+       Added optional  `return_counts`, `return_masks`, and `assume_unsorted`
+       parameters and changed dtype from ``np.int64`` to ``np.intp``
+       (corresponds to atom indices).
+    """
+    cdef np.intp_t n_values = values.shape[0]
+    cdef np.intp_t n_unique
+    cdef np.ndarray[np.intp_t, ndim=1] counts
+    cdef np.ndarray[object, ndim=1] masks
+    cdef np.ndarray[np.intp_t, ndim=1] unique = np.empty(n_values,
```

Comment:
old buffer interfaces on lines above

Reply:
That's here for the same reasons as above: `counts`, `masks`, and `unique` are the arrays being returned in the end and we want to avoid unnecessary calls to `np.asarray()`. Note that the low-level implementations of `unique_int_1d()` operate on memoryviews of these buffers.